Formula/kcptun.rb

```diff
@@ -0,0 +1,109 @@
+require "language/go"
+
+class Kcptun < Formula
+  desc "extremely simple & fast udp tunnel based on kcp protocol"
+  homepage "https://github.com/xtaci/kcptun"
+  url "https://github.com/xtaci/kcptun/archive/v20160701.tar.gz"
+  sha256 "865ad3dccd5bdd96075e7653da3b8ff4aaff576413c3b4e330d993e6f5209f68"
+  head "https://github.com/xtaci/kcptun.git"
+
+  depends_on "go" => :build
+
+  go_resource "golang.org/x/crypto" do
+    url "https://github.com/golang/crypto.git"
+  end
+
+  go_resource "github.com/golang/snappy" do
+    url "https://github.com/golang/snappy.git"
+  end
+
+  go_resource "github.com/hashicorp/yamux" do
+    url "https://github.com/hashicorp/yamux.git"
+  end
+
+  go_resource "github.com/urfave/cli" do
+    url "https://github.com/urfave/cli.git"
+  end
+
+  go_resource "github.com/xtaci/kcp-go" do
+    url "https://github.com/xtaci/kcp-go.git"
+  end
+
+  go_resource "github.com/klauspost/crc32" do
+    url "https://github.com/klauspost/crc32.git"
+  end
+
+  go_resource "github.com/klauspost/reedsolomon" do
+    url "https://github.com/klauspost/reedsolomon.git"
+  end
+
+  go_resource "github.com/klauspost/cpuid" do
+    url "https://github.com/klauspost/cpuid.git"
+  end
+
+  go_resource "golang.org/x/net" do
+    url "https://github.com/golang/net.git"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/xtaci"
+    ln_sf buildpath, buildpath/"src/github.com/xtaci/kcptun"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-ldflags", "-X main.VERSION=#{version} -s -w",
+      "-o", "kcptun_client", "github.com/xtaci/kcptun/client"
+    bin.install "kcptun_client"
+    (bin/"kcptun").write <<-EOS.undent
+      #!/bin/sh
+      #{bin}/kcptun_client -r serverip:554 -l 127.0.0.1:1080 -mode fast2
```

Comment:
There are a binary and a bash file. 
The bash file just exec binary file with args, it not must. But if you want to use `brew service`, it is needed.
So, i think the bash file should be in `bin` dir.